docs/references/contact-self.md

```diff
@@ -2,120 +2,158 @@
   * [intance](contact-self.md#contactself)
-    * [contactSelf.avatar\(\[file\]\) ⇒ `Promise <void | FileBox>`](contact-self.md#contactselfavatarfile-⇒-promise)
-    * [contactSelf.qrcode\(\) ⇒ `Promise <string>`](contact-self.md#contactselfqrcode-⇒-promise)
-    * [contactSelf.signature\(signature\) ⇒ `Promise <string>`](contact-self.md#contactselfsignaturesignature)
-    * [contactSelf.name\(\[name\]\) ⇒ `Promise <void> | string`](contact-self.md#contactselfname-⇒-promisestring)
+    * [contactSelf.avatar\(\[file\]\) ⇒ `None | FileBox`](contact-self.md#contactselfavatarfile-⇒-promise)
+    * [contactSelf.qrcode\(\) ⇒ `str`](contact-self.md#contactselfqrcode-⇒-promise)
+    * [contactSelf.signature\(signature\) ⇒ `str`](contact-self.md#contactselfsignaturesignature)
+    * [contactSelf.name\(\[name\]\) ⇒ `None | str`](contact-self.md#contactselfname-⇒-promisestring)
 
-### contactSelf.avatar\(\[file\]\) ⇒ `Promise <void | FileBox>`
+### contactSelf.avatar\(\[file\]\) ⇒ `None | FileBox`
 
-GET / SET bot avatar
+获取/设置 机器人所使用账号的头像
 
-**Kind**: instance method of [`ContactSelf`](contact-self.md#ContactSelf)
+**类型**: [`ContactSelf`](contact-self.md#ContactSelf)类的实例方法
 
-| Param | Type |
+| 参数 | 类型 |
 | :--- | :--- |
 | \[file\] | `FileBox` |
 
-**Example** _\( GET the avatar for bot, return {Promise&lt;FileBox&gt;}\)_
+**示例** 
 
-```javascript
-// Save avatar to local file like `1-name.jpg`
+_\( 获取机器人账号的头像, 返回`FileBox`类型的对象\)_
 
-bot.on('login', async user => {
-  console.log(`user ${user} login`)
-  const file = await user.avatar()
-  const name = file.name
-  await file.toFile(name, true)
-  console.log(`Save bot avatar: ${user.name()} with avatar file: ${name}`)
-})
+```python
+# 保存头像到本地文件, 类似 `1-name.jpg`的格式
+import asyncio
+from wechaty import Wechaty, FileBox, Contact
+
+class MyBot(Wechaty):
+
+    async def on_login(self, contact: Contact) -> None:
+        print(f"用户{contact}登入")
+        file: FileBox = await contact.avatar()
+        name = file.name
+        await file.to_file(name, True)
+        print(f"保存头像: {contact.name} 和头像文件: {name}")
+
+
+asyncio.run(MyBot().start())
 ```
 
-**Example** _\(SET the avatar for a bot\)_
+**示例** 
+
+_\(设置机器人账号的头像\)_
+
+```python
+import asyncio
+from wechaty import Wechaty, FileBox, Contact
 
-```javascript
-import { FileBox }  from 'file-box'
-bot.on('login', user => {
-  console.log(`user ${user} login`)
-  const fileBox = FileBox.fromUrl('https://wechaty.github.io/wechaty/images/bot-qr-code.png')
-  await user.avatar(fileBox)
-  console.log(`Change bot avatar successfully!`)
-})
+class MyBot(Wechaty):
+
+    async def on_login(self, contact: Contact) -> None:
+        print(f"用户{contact}登入")
+        file_box: FileBox = FileBox.from_url('https://wechaty.github.io/wechaty/images/bot-qr-code.png')
+        await contact.avatar(file_box)
+        print(f"更改账号头像成功")
+
+
+asyncio.run(MyBot().start())
 ```
 
-### contactSelf.qrcode\(\) ⇒ `Promise <string>`
+### contactSelf.qrcode\(\) ⇒ `str`
```

Comment:
The same as above suggestion that the result type annotation of async method should `Coroutine`.

Reply:
The example code is:

```python
async def qr_code(self) -> str
```
So the developer will get more detail about the reference.

Reply:
could you take a look at the new commit, is this format what we want?